### PR TITLE
Consolidate all `InlineFlowTest` libraries in the dataflow qlpack

### PR DESCRIPTION
**Pull request by jketema**

### CODE REVIEW

**Review by atorralba — APPROVED** [review]
Java changes LGTM. Thanks for doing this, it feels good to witness tests and test libraries getting increasingly cleaner :)

File: shared/dataflow/codeql/dataflow/test/InlineFlowTest.qll [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    int fieldFlowBranchLimit() { result = 1000 }
+  }
+
+  private module NoFlowConfig implements DataFlow::ConfigSig {
+    predicate isSource(DataFlowLang::Node source) { none() }
+
+    predicate isSink(DataFlowLang::Node sink) { none() }
+  }
+
+  module FlowTest<DataFlow::ConfigSig ValueFlowConfig, DataFlow::ConfigSig TaintFlowConfig> {
```
Comment (hvitved) [inline shared/dataflow/codeql/dataflow/test/InlineFlowTest.qll:65]: Can this be private?

File: go/ql/test/library-tests/semmle/go/dataflow/PromotedMethods/DataFlowConfig.ql [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import go
+import TestUtilities.InlineExpectationsTest
```
Comment (hvitved) [inline go/ql/test/library-tests/semmle/go/dataflow/PromotedMethods/DataFlowConfig.ql:2]: Why is this import needed?

File: java/ql/test/library-tests/dataflow/synth-global/test.ql [hunk 3]
```diff
@@ -1,4 +1,5 @@
 import java
+import semmle.code.java.dataflow.ExternalFlow
```
Comment (hvitved) [inline java/ql/test/library-tests/dataflow/synth-global/test.ql:2]: same

**Review by hvitved — APPROVED** [review]
Looks great, thanks for doing this. A few minor questions.

File: go/ql/test/library-tests/semmle/go/dataflow/PromotedMethods/DataFlowConfig.ql (hunk 2 above)
Comment (jketema) [reply go/ql/test/library-tests/semmle/go/dataflow/PromotedMethods/DataFlowConfig.ql:2]: This was originally exposed through `InlineFlowTest`, which it is no longer. It's needed later in this file, because it also defines an inline expectation test next to an inline flow test.

File: java/ql/test/library-tests/dataflow/synth-global/test.ql (hunk 3 above)
Comment (jketema) [reply java/ql/test/library-tests/dataflow/synth-global/test.ql:2]: This was originally exposed through `InlineFlowTest`, which it is no longer. It's needed later in this file for `ModelValidation`.

File: shared/dataflow/codeql/dataflow/test/InlineFlowTest.qll (hunk 1 above)
Comment (jketema) [reply shared/dataflow/codeql/dataflow/test/InlineFlowTest.qll:65]: No, there are tests that override both configurations. For example:

https://github.com/github/codeql/blob/3343b780154c5fafe9e378412697fb4c43e5ff3e/ruby/ql/test/library-tests/frameworks/sinatra/Flow.ql#L19

File: go/ql/test/library-tests/semmle/go/dataflow/PromotedMethods/DataFlowConfig.ql (hunk 2 above)
Comment (jketema) [reply go/ql/test/library-tests/semmle/go/dataflow/PromotedMethods/DataFlowConfig.ql:2]: I'd happily expose this again through `InlineFlowTest` (and the same below), but this seems somewhat cleaner.

Comment (hvitved) [reply go/ql/test/library-tests/semmle/go/dataflow/PromotedMethods/DataFlowConfig.ql:2]: Yeah, this is certainly better.
